chapters/batching.rst

```diff
@@ -0,0 +1,131 @@
+=============================================
+ Understanding Batching in Jina
+=============================================
+
+Some :term:`Executors` in Jina may profit from the capacity to process multiple instances of their input at the same time.
+This can lead to a speed up and higher throughput.
+
+In order to provide this capability, the :term:`Driver` extracts all the needed information from all the :term:`Document` in a :term:`Request`. [see `guide on request size <https://docs.jina.ai/chapters/request_size/>`_ ].
+
+Then all these instances (one per :term:`Document`) is passed to the :term:`Executor`. Then the :term:`Executor` decides how it wants to
+consume these instances (in batches or one by one).
+
+Batching decorators
+--------------------
+To control how an :term:`Executor` consumes its incoming inputs, Jina provides a set of decorator functions.
+
+single and single_multi_input
+------------------------------
+The single decorator ensures that even when the function is called with a set of instances, they will be processed one by one by the executor.
+This decorator is useful when the executor won't have benefits of processing inputs in batches, and allows the executor to keep a clean interface.
```

Comment:
Sorry, I'm still not sure I understand the benefits.
Does this(single decorator) means that we first have 1 batch of docs, then we can craft them into 1 by 1 list of docs?
What is the case when _the executor won't have benefits of processing inputs in batches_? What does it mean to keep a clean interface? Do we have an example if it's not a clean interface?

Reply:
So see the example there, it is not the same to have a simple instance ( a str ) than having a list of strings of size 1